WordPress/gutenberg
# Style Book: Show tabs and make blocks clickable when entering edit mode from the Styles menu

Fixes: #51887

## What?
This PR fixes three problems when entering edit mode with the Style Book open from the Styles menu.

- Canvas resizing is not possible
- Category tabs are not displayed
- Can't click blocks

![edit-style-book](https://github.com/WordPress/gutenberg/assets/54422211/5257e920-daf4-4eb3-8e69-49b10e3190e8)

## Why?
As I understand it, the Style Book displayed in the canvas area is based on the [`<GlobalStylesStyleBook />`](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/global-styles/ui.js#L243). In this component, the canvas is resizable, tabs are visible, and blocks are clickable (default behavior).

However, the Styles page overrides this with a Style Book in [`<SidebarNavigationScreenGlobalStyles />`](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js#L166) to [prohibit canvas resizing, hide tabs, and make the entire canvas clickable](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js#L250-L257).

On the Styles page, the URL contains `wp_global_styles`. I believe this is causing this issue because the Style Book in the `<SidebarNavigationScreenGlobalStyles />` [has priority](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar/index.js#L47-L49) even when entering edit mode from this page.

## How?
I used `getCanvasMode()` to not render the Style Book in the `<SizebarNavigationScreenGlobalStyles />` when in **edit** mode. This results in the Style Book being _reset_ to its closed state when you open the Style Book on the Styles page, enter edit mode, and return to view mode. Personally, I think this is acceptable, as it is the intended behavior.

## Testing Instructions

- Open the Site Editor and access the Styles menu.
- Click the Style Book button.
- The Style Book will appear on the canvas. There should be no category tab and all blocks should be displayed.
- Click the canvas area.
- The Style Book should have category tabs, clickable blocks, and a resizable canvas.
- Press the Open Navigation button to return to the Styles menu.
- The canvas area **should not show the Style Book, but the default template.**

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/76773d49-3f2b-4c4b-8541-a89f151edb8b
